can/CMakeLists.txt

```diff
@@ -5,6 +5,7 @@ else()
     if("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Linux")
         # Simulator requires linux only kernel interfaces
         add_subdirectory(simulator)
```

Comment:
Would it mess things up to make the simulator the static library here rather than creating a separate directory?

Reply:
I decided to not make a shared library. The reason is that the socket can code depended on FreeRTOS which is a pain using shared libraries due to the freertosconfig.h requirement.

I removed the dependency and made the socket can stuff templates.